### PR TITLE
Stack bar graph

### PR DESCRIPTION
**Pull request by whitetiger1399**
Screenshot from 2021-06-07 16-35-42

1. StackBarGraph component with bar and line graph added

### CODE REVIEW

File: src/lab/Graphs/StackBars/base.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+export interface BarDateValue {
+  // Date in milliseconds
+  date: number | string;
+  // Value to the corresponding date stamp
+  value: number | string;
+
+  runId?: string | undefined;
```
Comment (arkajyotiMukherjee) [inline src/lab/Graphs/StackBars/base.ts]: runId -> id

File: src/lab/Graphs/StackBars/base.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  // Array of {date and value}
+  data: Array<BarDateValue>;
+
+  // Color of the metric in the graph and legends
+  baseColor?: string;
+}
+
+export interface StackBarMetric {
+  runId: string;
```
Comment (arkajyotiMukherjee) [inline src/lab/Graphs/StackBars/base.ts]: same

File: src/lab/Graphs/StackBars/PlotStackBar.tsx [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+            width={xMax}
+            height={yMax}
+            onMouseMove={handleTooltip}
+            fill={"transparent"}
+            onClick={() => {
+              handleBarClick &&
+                handleBarClick(
+                  tooltipData
+                    ? tooltipData[tooltipData.length - 1].data.runId
+                    : ""
```
Comment (arkajyotiMukherjee) [inline src/lab/Graphs/StackBars/PlotStackBar.tsx]: can we provide null instead of ""

**Review by arkajyotiMukherjee — CHANGES_REQUESTED** [review]

**Review by arkajyotiMukherjee — APPROVED** [review]

**Review by vanshBhatia-A4k9 — APPROVED** [review]
- [ ] :rocket:

**Pull request merged**